src/h2olog/h2olog.cc

```diff
@@ -189,20 +198,30 @@ int main(int argc, char **argv)
         tracer.reset(create_http_tracer());
     }
 
+    std::vector<h2o_tracer::usdt> available_usdts = tracer->usdt_probes();
+
     int debug = 0;
+    bool list_and_exit = false;
     FILE *outfp = stdout;
-    std::vector<std::string> event_type_filters;
+    std::vector<h2o_tracer::usdt> event_type_filters;
     std::vector<std::string> response_header_filters;
     int c;
     pid_t h2o_pid = -1;
-    while ((c = getopt(argc, argv, "hdp:t:s:w:")) != -1) {
+    while ((c = getopt(argc, argv, "hdp:t:s:w:l")) != -1) {
         switch (c) {
         case 'p':
             h2o_pid = atoi(optarg);
             break;
-        case 't':
-            event_type_filters.push_back(optarg);
+        case 't': {
+            auto found = std::find_if(available_usdts.cbegin(), available_usdts.cend(),
+                                      [](const h2o_tracer::usdt &usdt) { return optarg == usdt.fully_qualified_name(); });
+            if (found == available_usdts.cend()) {
+                fprintf(stderr, "No such event type: %s\n", optarg);
```

Comment:
Would need to change "event type" to something else based on the outcome of the point above.